pkg/controller/controllerutil/controllerutil.go

```diff
@@ -334,6 +336,47 @@ func CreateOrPatch(ctx context.Context, c client.Client, obj client.Object, f Mu
 	return result, nil
 }
 
+// CreateIfNotExist creates the given object in the Kubernetes
+// cluster. The object's desired state must be reconciled with the existing
+// state inside the passed in callback MutateFn.
+//
+// It returns the executed operation and an error.
+func CreateIfNotExist(ctx context.Context, c client.Client, obj client.Object, f MutateFn) (OperationResult, error) {
```

Comment:
Not sure this is common enough to warrant a helper. Controllers really shouldn't be doing this except in very rare cases as it's a sign of non-convergent behavior.

Reply:
@coderanger we have a teams that use a variety of *IfNotExist methods. They created them because controller-runtime didn't have them. We distilled the methods do this specific `CreateIfNotExist` variant. We were hoping to let the teams know that controller-runtime now supports their use case and stop maintaining their variants. That's what brought this PR on.

Reply:
Could you explain the use cases that lead to needing them? I don't doubt that you have code that does it, what I doubt is if that code is a good idea and should be encouraged :)